### ISSUE TITLE
nvim-treesitter module support will be removed

### ISSUE THREAD

**Issue by cryptomilk** [RRethy/nvim-treesitter-endwise#27]
**Describe the bug**
nvim-treesiter module support will be removed

https://github.com/nvim-treesitter/nvim-treesitter/issues/4767

**Comment by brianhuster** [comment]
From nvim-treesitter/nvim-treesitter#4767

> after Neovim 0.11 is released with wasm support, main will become the new default branch

Neovim 0.11 will be released in mid March this year, and if you test this plugin with the "main" branch of nvim-treesitter/nvim-treesitter, it won't work. Because then nvim-treesitter will only be like a Treesitter parser manager, it will no longer provide any public utility function to work with Treesitter. 

So I think this issue should still be opened.

**Comment by stevenxxiu** [comment]
The `master` branch of `treesitter` is now archived. See <https://github.com/nvim-treesitter/nvim-treesitter/commit/42fc28ba918343ebfd5565147a42a26580579482>.

This plugin doesn't work with the `main` branch.

The `require("nvim-treesitter.query")` doesn't exist anymore.

**Comment by brianhuster** [comment]
@stevenxxiu you can try using my fork https://github.com/brianhuster/treesitter-endwise.nvim

**Comment by chuan29812** [comment]
@brianhuster thanks for the fork, yeah it works perfectly, hoping @RRethy will be able to update when time is available. cheers

**Comment by chuan29812** [comment]
edit: false alarm. @brianhuster fork works flawlessly

**Comment by brianhuster** [comment]
That's strange, my fork is not called nvim-treesitter-endwise

Also the plugin is not "dependency of treesitter"

**Comment by chuan29812** [comment]
> That's strange, my fork is not called nvim-treesitter-endwise
> 
> Also the plugin is not "dependency of treesitter"

Im dumb, i had the wrong url, but somehow was updated fine so i thought it must have worked. your branch is perfect, my bad.
And I also corrected the dependency part haha

**Comment by mkalinski** [comment]
> Im dumb, i had the wrong url, but somehow was updated fine so i thought it must have worked. 

The fork definitely *was* called `brianhuster/nvim-treesitter-endwise` when the original post was made 3 days ago, but was renamed since then for some reason.

It causes a bit of confusion when updating nvim plugins, because instead of saying "the repository no longer exists / was moved", Github inexplicably prompts for login (at least on my setup).

**Comment by brianhuster** [comment]
> The fork definitely was called brianhuster/nvim-treesitter-endwise when the original post was made 3 days ago, but was renamed since then for some reason.

I don't understand what you said. It has been called treesitter-endwise.nvim **_for months_** (since I felt like @RRethy was not interested in my proposal), not just "since 3 days ago". You are free to check the history of my comment 3 days ago that you refer to https://github.com/RRethy/nvim-treesitter-endwise/issues/27#issuecomment-2907884233

**Comment by mkalinski** [comment]
Then it has to be some Github glitch. I have checked out *something*, but it seems to be neither this repo, nor your current repo, but instead some outdated fork?

![Image](https://github.com/user-attachments/assets/8988d172-ad7f-4cf3-8992-fca659e5f487)

Anyway, it's not important. Sorry for the offtopic, I just thought the rename was recent, and my comment might help someone who has sudden problems with updating the plugin.

**Comment by chuan29812** [comment]
> Then it has to be some Github glitch. I have checked out _something_, but it seems to be neither this repo, nor your current repo, but instead some outdated fork?
> 
that explains it, i changed it to the fork, and manually updated it `endwise` twice in Lazy, never complained about unable to find it. Glad its all sorted out now

**Comment by Bekaboo** [comment]
@brianhuster Thanks for your great work. Would you consider upstreaming your fixes to this repo?

**Comment by chuan29812** [comment]
Just throwing this out there, anyone having very occasional problems with errors like `cant find parser for buffer`?

**Comment by brianhuster** [comment]
> Thanks for your great work. Would you consider upstreaming your fixes to this repo?

Yes, I did, but upstream currently even supports Nvim < 0.9, while I only want to support Nvim 0.10+ (which is still lower requirement than the one of `nvim-treesitter`'s `main` branch). However, I could upstream my new test workflow first, since it is probably lower-risk.

On the other hand, @RRethy is free to take any parts of my fork to upstream.

**Comment by RRethy** [comment]
Probably won't have time to implement a solution here for until week of June 23rd - June 27th. I'm hesitant to accept any PRs since I need to fully understand the code before merging it. If people can't wait, feel free to switch to https://github.com/brianhuster/treesitter-endwise.nvim, and when I ship a fix you can always switch back if you want.

**Comment by Bekaboo** [comment]
@RRethy Gentle ping 🙂 Is there any plan to adapt to nvim-treesitter `main` in recent days?

**Comment by brianhuster** [comment]
> I'm hesitant to accept any PRs since I need to fully understand the code before merging it

Supporting the `main` branch of [nvim-treesitter](https://github.com/nvim-treesitter/nvim-treesitter) is quite simple, you just need to replace any `nvim-treesitter` API call with corresponding `vim.treesitter` API (or implementing your own when there is none). So there won't be much code you need to "fully understand".
